Cython/Compiler/Builtin.py

```diff
@@ -496,6 +511,8 @@ def init_builtins():
     complex_type  = builtin_scope.lookup('complex').type
     memoryview_type = builtin_scope.lookup('memoryview').type
 
+    sequence_types = (list_type, tuple_type, bytes_type, str_type, unicode_type, basestring_type, bytearray_type)
```

Comment:
One point that might get confusing: this doesn't quite correspond to the types in `collections.abc.Sequence`, which should also have `memoryview` in. Although `memoryview` wouldn't work with this optimization I think.

Reply:
I added memoryview (for completeness) and excluded it explicitly in `MulNode`. Feels safer to do that.